src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DynamicFiles/IDynamicDocumentContainer.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+
+namespace Microsoft.CodeAnalysis.Razor.DynamicFiles;
+
+internal interface IDynamicDocumentContainer
+{
+    string FilePath { get; }
+
+    bool SupportsDiagnostics { get; set; }
```

Comment:
Yeah, this is particularly icky. I didn't take a hard look at this at the time because I was just making a simple transformation. When I saw that there was code that sets it, I left it in place.

I'll dig in further to see if there's something better that can be done.

Reply:
Unfortunately, the value of `SupportsDiagnostics` *does* vary and is set. However, I can convert the setter to a method, which makes it feel a bit better.